command/sub8_missions/missions/arm_torpedos.py

```diff
@@ -0,0 +1,158 @@
+
+        self.pattern_done = False
+        for i, move in enumerate(self.moves[self.move_index:]):
+            move = self.sub.move.relative(np.array(move)).go(blind=self.BLIND)
+            move.addErrback(err)
+            yield move
+            self.move_index = i + 1
+        self.print_bad('Pattern finished. Firing at any locked targets.')
+        self.pattern_done = True
+
+    @util.cancellableInlineCallbacks
+    def fire(self, target):
+        self.print_info("FIRING {}".format(target))
+        yield self.sub.move.go(blind=self.BLIND)  # Station hold
+        target_position = self.targets[target].position
+        yield self.sub.move.depth(-target_position[2]).go(blind=self.BLIND)
+        yield self.sub.move.look_at_without_pitching(target_position).go(blind=self.BLIND)
+        yield self.sub.move.set_position(target_position).backward(1).go(blind=self.BLIND)
+        self.print_good(
+            "{} locked. Firing torpedos. Hit confirmed, good job Commander.".format(target))
+        self.done = True
+
+    def get_target(self):
+        target = 'BCO'
+        '''
+        Returns the target we are going to focus on. Loop through priorities.
+        Highest priority is the TCX target, followed by the TRX and TLX.
+        Lowest priority is the BCO target. Targets are ordered in the list already,
+        so this will take the first target it can actually find. Currently this
+        will always by the BCO target. This is because we don't differentiate
+        between targets currently as we cannot unblock the blocked ones.
+        This means there is only one target for us to lock onto. 
+        '''
+        if self.targets[target].destroyed:
+            # continue
+            temp = target
+        elif self.targets[target].position is not None:
+            return target
+        elif self.pattern_done and self.targets[target].position is not None:
+            return target
+        else:
+            return None
+
+    @util.cancellableInlineCallbacks
+    def run(self):
+        # start_time = yield self.sub.nh.get_time()  # Store time mission
+        # starts for timeout
+
+        self.print_info("Enabling Perception")
+        self.sub.vision_proxies.arm_torpedos.start()
+
+        pattern = self.pattern()
+        self.do_search = True
+        search = self.search()
+        while not self.done:
+            t = self.get_target()
+            if t is not None:
+                pattern.cancel()
+                yield self.fire(t)
+                self.targets[t].set_destroyed()
+                if not self.done:
+                    pattern = self.pattern()
+            elif self.pattern_done:
+                break
+            yield self.sub.nh.sleep(0.1)
+        search.cancel()
+        pattern.cancel()
+        self.print_good('Done!')
+
+
+@util.cancellableInlineCallbacks
+def run(sub):
+    # print('running')
+    mission = FireTorpedos(sub)
+    yield mission.run()
```

Comment:
Looks fine, but I'm concerned about the situation where the target goes out of frame